mesa/visualization/backends/altair_backend.py

```diff
@@ -0,0 +1,598 @@
+
+                aps = AgentPortrayalStyle(
+                    x=agent_x,
+                    y=agent_y,
+                    size=dict_data.pop("size", style_fields.get("size")),
+                    color=dict_data.pop("color", style_fields.get("color")),
+                    marker=dict_data.pop("marker", style_fields.get("marker")),
+                    zorder=dict_data.pop("zorder", style_fields.get("zorder")),
+                    alpha=dict_data.pop("alpha", style_fields.get("alpha")),
+                    edgecolors=dict_data.pop(
+                        "edgecolors", style_fields.get("edgecolors")
+                    ),
+                    linewidths=dict_data.pop(
+                        "linewidths", style_fields.get("linewidths")
+                    ),
+                )
+                if dict_data:
+                    ignored_keys = list(dict_data.keys())
+                    warnings.warn(
+                        f"The following keys were ignored from dict portrayal: {', '.join(ignored_keys)}",
+                        UserWarning,
+                        stacklevel=2,
+                    )
+            else:
+                aps = portray_input
+                if aps.x is None and aps.y is None:
+                    aps.x, aps.y = self._get_agent_pos(agent, space)
+
+            arguments["loc"].append((aps.x, aps.y))
+
+            size_to_collect = aps.size if aps.size is not None else default_size
+            if size_to_collect is None:
+                size_to_collect = class_default_size
+            arguments["size"].append(size_to_collect)
+
+            arguments["color"].append(
+                aps.color if aps.color is not None else style_fields.get("color")
+            )
+
+            # Map marker to Altair shape if defined, else use raw marker
+            raw_marker = (
+                aps.marker if aps.marker is not None else style_fields.get("marker")
+            )
+            shape_value = marker_to_shape_map.get(raw_marker, raw_marker)
+            if shape_value is None:
+                warnings.warn(
+                    f"Marker '{raw_marker}' is not supported in Altair. "
+                    "Using 'circle' as default.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+                shape_value = "circle"
+            arguments["shape"].append(shape_value)
+
+            arguments["order"].append(
+                aps.zorder if aps.zorder is not None else style_fields.get("zorder")
+            )
+            arguments["opacity"].append(
+                aps.alpha if aps.alpha is not None else style_fields.get("alpha")
+            )
+            arguments["stroke"].append(aps.edgecolors)
+            arguments["strokeWidth"].append(
+                aps.linewidths
+                if aps.linewidths is not None
+                else style_fields.get("linewidths")
+            )
+
+            # FIXME: Kind of stupid logic.
```

Comment:
Can you add some more clarification here in case someone else wants to do a PR on the fix